internal/resourcepolicies/resource_policies_mgr.go

```diff
@@ -0,0 +1,139 @@
+package resourcepolicies
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kbClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// resPoliciesMeta stored one resource policies and related resource policies configmap creation timestamp
+type resPoliciesMeta struct {
+	policies                *ResourcePolicies
+	configCreationTimestamp metav1.Time
+}
+
+// resPoliciesKey identify one specific resPoliciesMeta with refName and refType
+type resPoliciesKey struct {
+	refName string
+	refType string
+}
+
+// ResPoliciesManager manage all resource policies in memory and keep all up-to-date
+type ResPoliciesManager struct {
+	resPolicies map[resPoliciesKey]*resPoliciesMeta
+	namespace   string
+	kbClient    kbClient.Client
+}
+
+var ResPoliciesMgr ResPoliciesManager
+
+func (mgr *ResPoliciesManager) InitResPoliciesMgr(namespace string, kbClient kbClient.Client) {
+	mgr.resPolicies = make(map[resPoliciesKey]*resPoliciesMeta)
+	mgr.namespace = namespace
+	mgr.kbClient = kbClient
+}
+
+// getResourcePolicies first check and update current resource policies and then get the latest resource policies
+func (mgr *ResPoliciesManager) getResourcePolicies(refName, refType string) (*ResourcePolicies, error) {
```

Comment:
How about refactoring the flow of this method as follows?
1. Get the `configmap` from the API server and check whether it's corresponding policy already exists. If it does, return directly
2. If not, build the policy, put it into the map and return

Reply:
ok, I'll modify it

Reply:
updated